dot/network/host.go

```diff
@@ -437,3 +438,38 @@ func (h *host) closeProtocolStream(pID protocol.ID, p peer.ID) {
 		}
 	}
 }
+
+// Process will connect, drop or reject a peer based on a peerset message
+func (h *host) Process(msg peerset.Message) {
+	peerID := msg.PeerID
+	if peerID == "" {
+		logger.Errorf("found empty peer id in peerset message")
+		return
+	}
+	switch msg.Status {
+	case peerset.Connect:
+		addrInfo := h.h.Peerstore().PeerInfo(peerID)
+		if len(addrInfo.Addrs) == 0 {
+			var err error
+			addrInfo, err = h.discovery.findPeer(peerID)
+			if err != nil {
+				logger.Warnf("failed to find peer id %s: %s", peerID, err)
+				return
+			}
+		}
+
+		err := h.connect(addrInfo)
+		if err != nil {
+			logger.Warnf("failed to open connection for peer %s: %s", peerID, err)
+			return
+		}
+		logger.Debugf("connection successful with peer %s", peerID)
+	case peerset.Drop, peerset.Reject:
+		err := h.closePeer(peerID)
```

Comment:
Does closePeer () handle the case where a peer isn't connected (imagining thats the Reject case)?

Reply:
Right now Reject case is processed but not the right way I think. The `network.handleConn(conn libp2pnetwork.Conn)` which started the incoming calls the `peersetHandler.Incoming` method which should check if the peer should be accepted or rejected but this answer is sent to the processor instead of returned to the `network.handleConn` to stop the handshake with the peer.

I think we should change the following method:

````
func (s *Service) handleConn(conn libp2pnetwork.Conn) {
	// TODO: currently we only have one set so setID is 0, 
        // change this once we have more set in peerSet.
	s.host.cm.peerSetHandler.Incoming(0, conn.RemotePeer())
```` 

to 
````
func (s *Service) handleConn(conn libp2pnetwork.Conn) {
	// TODO: currently we only have one set so setID is 0, 
        // change this once we have more set in peerSet.
	result := s.host.cm.peerSetHandler.Incoming(0, conn.RemotePeer())
        switch result {
        case Reject: 
            logger.Infof("reject ...")
            return
        }

        // exchange BlockAnnounceHandshake with peer so we can start to
	// sync if necessary.
	prtl, has := s.notificationsProtocols[BlockAnnounceMsgType]
	if !has {
		return
	}

	hs, err := prtl.getHandshake()
        ...
```` 
@noot, @timwu20 what do you think?

Reply:
@EclesioMeloJunior that makes sense, also in the Reject case you would need to call `conn.Close()` to close the connection with the peer